tower-http/src/sensitive_header.rs

```diff
@@ -236,32 +273,36 @@ where
     fn call(&mut self, req: Request<ReqBody>) -> Self::Future {
         SetSensitiveResponseHeaderResponseFuture {
             future: self.inner.call(req),
-            header: self.header.clone(),
+            headers: Some(self.headers.clone()),
```

Comment:
For response headers, I am not sure if there is as good a way to avoid cloning the headers, like we can for the request headers, since they have to be passed into the response future in any case.

However, if we used the 
```rust
    for<'a> &'a I: IntoIterator<Item = &'a HeaderName>
```
as I discussed above, it would at least permit using an `Arc<Vec<HeaderName>>` in this case, which would make this clone much more lightweight than cloning an entire `Vec` into every response future. In fact, I think we _might_ want to remove the `Clone` bound on `I` and just always `Arc` it, for the response headers case?

That might add an unnecessary `Arc` in the case where `I` is `Option<HeaderName>` and the `HeaderName` was constructed with `HeaderName::from_static` or similar...but that might be worthwhile if it makes it impossible to accidentally write code that clones a large `Vec<HeaderName>` into every response future...